server/src/main/java/com/linecorp/centraldogma/server/internal/httpapi/ContentServiceV1.java

```diff
@@ -0,0 +1,334 @@
+        final String summary = commitMessage.getSummary();
+        final String detail = commitMessage.getDetail();
+        final Markup markup = Markup.getMarkup(commitMessage.getMarkup());
+
+        return repository.normalize(revision).thenCompose(normalizedRevision -> execute(
+                Command.push(commitTimeMills, author, repository.parent().name(), repository.name(),
+                             normalizedRevision,
+                             summary, detail, markup, changes)));
+    }
+
+    private Repository getRepository(String projectName, String repoName) {
+        return projectManager().get(projectName).repos().get(repoName);
+    }
+
+    /**
+     * GET /projects/{projectName}/repositories/{repository}/contents{path}?revision={revision}
+     * queryType={queryType}&expression={expression}
+     *
+     * <p>Returns the entry of files in the path. This is same with
+     * {@link #listFiles(String, String, String, String)} except that containing the content of the files.
+     */
+    @Get("regex:/projects/(?<projectName>[^/]+)/repos/(?<repoName>[^/]+)/contents(?<path>/.*$)")
+    public CompletionStage<?> getFiles(@Param("projectName") String projectName,
+                                         @Param("repoName") String repoName,
+                                         @Param("path") String path,
+                                         @Param("revision") @Default("-1") String revision,
+                                         @Param("queryType") Optional<String> queryType,
+                                         @Param("expression") Optional<String> expressions) {
+        checkRepositoryExists(projectName, repoName);
+
+        if (!path.endsWith("/") && !path.endsWith("*")) {
+            final Query<?> query = Query.of(
+                    QueryType.valueOf(queryType.orElse("IDENTITY")), path, expressions.orElse(""));
+            if (query.type() == QueryType.JSON_PATH && !toLowerCase(path).endsWith(".json")) {
+                throw new HttpResponseException(HttpStatus.BAD_REQUEST);
+            }
+
+            final Repository repository = projectManager().get(projectName).repos().get(repoName);
+            return repository.get(new Revision(revision), query).thenApply(
+                    queryResult -> DtoConverter.convert(queryResult, path));
+        } else {
+            if (queryType.isPresent() && toLowerCase(queryType.get()).equalsIgnoreCase("json_path")) {
+                throw new HttpResponseException(HttpStatus.BAD_REQUEST);
+            }
+            final Repository repository = projectManager().get(projectName).repos().get(repoName);
+            return listFiles(repository, path, new Revision(revision), ImmutableMap.of());
+        }
+    }
+
+    /**
+     * DELETE /projects/{projectName}/repos/{repoName}/contents{path}
+     *
+     * <p>Deletes a file.
+     */
+    @Delete("regex:/projects/(?<projectName>[^/]+)/repos/(?<repoName>[^/]+)/contents(?<path>/.*$)")
```

Comment:
Do we need convenience methods for this on upstream? like HttpClient.delete(uri, string body)? It is not a usual case though.

Reply:
Like adding `Repository.deleteFiles(Revision revision, String pathPattern)`? Sounds good to me.